src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1069,6 +1075,46 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
             }
         }
 
+        internal static string GetDefaultHostId(ScriptSettingsManager settingsManager, ScriptHostConfiguration scriptConfig)
+        {
+            // We're setting the default here on the newly created configuration
+            // If the user has explicitly set the HostID via host.json, it will overwrite
+            // what we set here
+            string hostId = null;
+            if (scriptConfig.IsSelfHost)
```

Comment:
Rather than always fall back to doing this, I chose to scope it to CLI/SelfHost scenarios because this machine name approach won't work for scale out, since the ID needs to be the same for ALL instances.